src/lib/Libutil/misc_utils.c

```diff
@@ -715,6 +715,66 @@ replace(char *str, char	*sub, char *repl, char	*retstr)
 	strncpy(retstr, rstr, i + 1);
 }
 
+
+/**
+ * @brief
+ *	Add esc as escape character if delim is appearing on str
+ *
+ * @param[in]	str		- input string
+ * @param[in]	delim  	- delimiter to be searched on str
+ * @param[in]   esc 	- escape character to be added if delim found on str
+ * @param[out]  retstr 	- delim replaced with and escape character+delim if found.
```

Comment:
I believe retstr is an [in, out] param. Description could be more meaningful - for example, output string, with every occurence of delim escaped with 'esc'
Also, above 4 lines do not have uniform indentation.